docs/proposals/20220125-ipam-integration.md

```diff
@@ -0,0 +1,410 @@
+
+```yaml
+apiVersion: ipam.cluster.x-k8s.io/v1alpha1
+kind: InfobloxIPPool
+metadata:
+  name: ib-pool
+spec:
+  networkView: "some-view"
+  dnsZone: "test.example.com."
+  network: 10.10.10.0/24
+```
+
+### Consumption
+
+The consumers of this API contract are the infrastructure providers. As mentioned in the introduction, not all providers will require integration with IPAM systems. In addition, DHCP may be a viable or even better option in some environments. Integration with this API should therefore be implemented as an optional feature.
+
+In order to consume IPAddresses, infrastructure providers need to create an IPAddressClaim for each address they require, and then have to wait until all claims are fulfilled. When creating those claims, a deterministic naming scheme should be used, so it's easy to identify which claims belong to which machine, e.g. `<infra machine name>-<interface name>`.
+
+Since IPAddressClaims have to reference a Pool, the infrastructure provider needs to allow specifying a Pool in its infrastructure templates, and the referenced Pool also has to be created (or exist already) when creating a cluster. The pool should be in the same namespace.
+
+![relationships of ipam custom resources](images/ipam-integration/consumption.png)
+
+When creating an IPAddressClaim, the provider should set an owner reference to the infrastructure machine, so IPAM providers can fetch further information if necessary.
+
+Using CAPV as an example, a template *could* look like this:
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: VSphereMachineTemplate
+metadata:
+ name: example
+ namespace: vsphere-site1
+spec:
+ template:
+   spec:
+     cloneMode: FullClone
+     numCPUs: 8
+     memoryMiB: 8192
+     diskGiB: 45
+     network:
+       devices:
+       - dhcp4: false
+         fromPool: # reference to the pool
+           group: ipam.cluster.x-k8s.io/v1alpha1
+           kind: IPPool
+           name: testpool
+```
+
+A Machine generated from that template could then look like this:
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: VSphereMachine
+metadata:
+name: example-1
+namespace: vsphere-site1
+spec:
+ cloneMode: FullClone
+ numCPUs: 8
+ memoryMiB: 8192
+ diskGiB: 45
+ network:
+   devices:
+   - dhcp4: false
+     fromPool: # reference to the pool
+       group: ipam.cluster.x-k8s.io/v1alpha1
+       kind: IPPool
+       name: testpool
+status:
+ network:
+   devices:
+     - claim:
+         name: example-1-1
+```
+
+### User Stories
+
+- Detail the things that people will be able to do if this proposal is implemented.
+- Include as much detail as possible so that people can understand the "how" of the system.
+- The goal here is to make this feel real for users without getting bogged down.
+
+#### Story 1
+
+As a CAPI user I want to use static IP addresses with my Machines to avoid problems with DHCP lease duration during rolling cluster upgrades.
```

Comment:
can we be specific about which persona is the CAPI user here and the other stories? e.g cluster operator